gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/AllergyIntoleranceMapper.java

```diff
@@ -90,12 +96,29 @@ private AllergyIntolerance mapAllergyIntolerance(RCMRMT030101UK04EhrExtract ehrE
     }
 
     private void buildParticipantReferences(RCMRMT030101UK04EhrComposition ehrComposition,
-        RCMRMT030101UK04CompoundStatement compoundStatement, AllergyIntolerance allergyIntolerance) {
-        var practitioner = Optional.ofNullable(getParticipantReference(compoundStatement.getParticipant(), ehrComposition));
+                                            RCMRMT030101UK04CompoundStatement compoundStatement,
+                                            AllergyIntolerance allergyIntolerance) {
+
+        var recorderAndAsserter = fetchRecorderAndAsserter(ehrComposition);
+
+        if (recorderAndAsserter.get(RECORDER).isPresent() && recorderAndAsserter.get(AUTHOR).isPresent()) {
+            allergyIntolerance
+                        .setRecorder(recorderAndAsserter.get(RECORDER).get())
```

Comment:
I think the Java dev team recommend using `.orElseThrow` instead of `.get` for Optional as it more accurately describes what is happening.

Reply:
I think it is better to keep it this way as there are two conditions that we are checking here hence it is not very convenient and practical to use orElseThrow as if recorder or author don't exist it is not an exception in this case. It just simply goes into else block where both recorder and asserter should be populated with the same ID from the same participant.

Reply:
`get()` would throw if the `isPresent()` check wasn't done. I think this is the reason `orElseThrow()` is preferred. I also think `recorderAndAsserter.get(RECORDER).orElseThrow()` is easier to read. I understand the functionality is the same, I just wanted to point out what I believe is best practice. 

I've seen this mentioned by the Java dev team, it's also mentioned in [this](https://www.youtube.com/watch?v=V7iW27F9oog) video at 7:30. 

As it is just style and not functionality I'm happy for it to be merged without this change.